aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLOperation.java

```diff
@@ -142,6 +143,11 @@ public void onResponse(@NonNull Call call, @NonNull Response response) {
                     return;
                 }
             }
+            if (response.code() >= START_OF_CLIENT_ERROR_CODE && response.code() <= END_OF_CLIENT_ERROR_CODE) {
+                onFailure.accept(new ApiException
+                        .NonRetryableException("OkHttp client request failed.", "Irrecoverable error")
+                );
+            }
```

Comment:
Shouldn't the client have access to jsonResponse?  Maybe we can call onFailure instead of throwing exception?

Reply:
We are converting the client error code family to non retryable exception which is passed in to on failure call back. The exception is not thrown here.